### PR TITLE
Rz logging dedup2

### PR DESCRIPTION
**Pull request by rzinke**
Latest updates to prevent or minimize reprocessing for ariaTSsetup and ariaExtract.
Logging included.

### CODE REVIEW

**Comment by rzinke** [comment]
Test using
```
#!/bin/bash

clear

workdir=test

# Download products
ariaDownload.py -t 64 -b '34.5 35.7 -116.5 -114.2' -s 20180101 -e 20180201 -o download

# Extract products
ariaTSsetup.py -f "products/*.nc" -w $workdir --verbose --log-level debug

# Download more products
ariaDownload.py -t 64 -b '34.5 35.7 -116.5 -114.2' -s 20180101 -e 20180215 -o download

# Extract additional products
ariaTSsetup.py -f "products/*.nc" -w $workdir --verbose --log-level debug

# Run again to see how much faster
ariaTSsetup.py -f "products/*.nc" -w $workdir --verbose --log-level debug


workdir='test-aoi'

# Extract products
ariaTSsetup.py -f "products/*.nc" -b aoi.geojson -w $workdir --verbose --log-level debug

# Extract products again
ariaTSsetup.py -f "products/*.nc" -b aoi.geojson -w $workdir --verbose --log-level debug
```

File: tools/bin/ariaTSsetup.py [hunk 1]
```diff
@@ -475,7 +483,7 @@ def main():
     # Pass DEM-filename, loaded DEM array, and lat/lon arrays
     LOGGER.info('Download/cropping DEM')
     demfile, demfile_expanded, lat, lon = \
-        ARIAtools.util.dem.prep_dem(**dem_dict)
+        ARIAtools.util.dem.prep_dem(**dem_dict, run_log=run_log)
```
Comment (ehavazli) [inline tools/bin/ariaTSsetup.py:487]: this line causes the `ariaTSsetup.py` to die with the error:
`TypeError: prep_dem() got an unexpected keyword argument 'run_log'`

`run_log` needs to be added as a variable at the top of the function or removed from here.

File: tools/ARIAtools/util/run_logging.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import datetime
+import shapely
+
+import ARIAtools
+import ARIAtools.util.shp
+
+
+class RunLog:
+    """
+    """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:14]: populate or remove the docstring triple quotes

File: tools/ARIAtools/util/run_logging.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    def __init__(self, workdir, verbose=None):
+        """
+        """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:17]: populate or remove the docstring triple quotes

File: tools/ARIAtools/util/run_logging.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+                json.dump({}, extr_file)
+
+        # Automatically update core parameters
+        self.__update_runtimes__()
+        self.update('workdir', self.workdir)
+        self.update('update_mode', 'full_extract')
+
+    def load(self):
+        """
+        """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:55]: populate or remove the docstring triple quotes

File: tools/ARIAtools/util/run_logging.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        with open(self.log_name, 'rb') as log_file:
+            log_data = pickle.load(log_file)
+
+        return log_data
+
+    def update(self, atr_name, atr_value):
+        """
+        """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:63]: populate or remove the docstring triple quotes

File: tools/ARIAtools/util/run_logging.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+            self.__write_extracted_files__(atr_value)
+
+        # Write new log data
+        log_data[atr_name] = atr_value
+        with open(self.log_name, 'wb') as log_file:
+            pickle.dump(log_data, log_file)
+
+    def __update_runtimes__(self):
+        """
+        """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:102]: populate or remove the docstring triple quotes

File: tools/ARIAtools/util/run_logging.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        if 'run_times' in log_data.keys():
+            log_data['run_times'].append(self.run_time)
+        else:
+            log_data['run_times'] = [self.run_time]
+
+        self.update('run_times', log_data['run_times'])
+
+    def __update_configs__(self, atr_name, atr_value):
+        """
+        """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:115]: populate or remove the docstring triple quotes

File: tools/ARIAtools/util/run_logging.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+            config_data = json.load(config_file)
+
+        # Write new data
+        config_data[atr_name] = atr_value
+        with open(self.config_name, 'w') as config_file:
+            json.dump(config_data, config_file)
+
+    def __write_file_list__(self, files):
+        """
+        """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:131]: populate or remove the docstring triple quotes

File: tools/ARIAtools/util/run_logging.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+        # Write new data
+        with open(self.file_list_name, 'w') as list_file:
+            json.dump({'files': files}, list_file)
+
+    def __write_extracted_files__(self, extracted_files):
+        """
+        """
```
Comment (ehavazli) [inline tools/ARIAtools/util/run_logging.py:138]: populate or remove the docstring triple quotes

**Review by ehavazli — CHANGES_REQUESTED** [review]

**Comment by alexfore** [comment]
These are my comments

IMO should phase out use of verbose and let the log level control this type of messaging to the user.  The prints which are guarded by if verbose should probably be LOGGER.debug messages.

Recommend to rename run_logging module to runlog (shorter, module name matches class name i.e. runlog.RunLog).

use full namespace to match existing style in imports, almost always with very few exceptions -- this is why it is important to make short and explicit module and package names as per comment above.

imports in run_logging should roughly follow the isort ordering (i.e. in three blocks: system, third party, our own)

doc strings should be populated with something

Ideally, logger messages should use lazy formatting, not f-strings or % strings. These are formatted even if the logger message is not written, whereas lazy formatted strings are only formatted if the message will be written. Changing it everywhere would be a premature optimization but one should get in the habit of using lazy formatting always going forward.

recommend renaming atr to attr in atr_name, atr_value. This is more pythonic -- see getattr / setattr method names.

File: tools/ARIAtools/extractProduct.py [hunk 10]
```diff
@@ -474,6 +511,48 @@ def merged_productbbox(
...
+            print(f'Overlap ratio (new/prev): {olap_ratio:.7f}')
+
+        if (delta_area != 0.0) or (olap_ratio != 1.0):
+            LOGGER.debug(f'Product bbox changed in size from previous run '
+                         f'{new_area} vs {exist_area}')
+
+        if shapely.equals(new_bbox, exist_bbox):
+            # Same bbox within machine precision
+            update_mode = 'skip'
+        elif olap_ratio < 1.0:
```
Comment (mgovorcin) [inline tools/ARIAtools/extractProduct.py:545]: we should have some warning message if olap ration is below some threshold, e.g. 0.6 (60%) indicating drastic change in aoi

File: tools/ARIAtools/extractProduct.py [hunk 11]
```diff
@@ -474,6 +511,48 @@ def merged_productbbox(
...
+        olap_area = ARIAtools.util.shp.shp_area(olap_bbox, lyr_proj)
+
+        # Compare areas
+        delta_area = np.abs(olap_area - exist_area)
+        delta_area = np.round(delta_area * 1E7) * 1E-7
+
+        olap_ratio = olap_area / exist_area
+        olap_ratio = np.round(olap_ratio * 1E7) * 1E-7
+
+        if verbose:
```
Comment (mgovorcin) [inline tools/ARIAtools/extractProduct.py:533]: agree with Alex to replace verbose with logger

File: tools/ARIAtools/extractProduct.py (hunk 10 above)
Comment (mgovorcin) [inline tools/ARIAtools/extractProduct.py:545]: would be good to add loggin.WARN if olap_ration falls below some threshold, e.g. 0.6 (60% change0 to warn about significantly smaller area

File: tools/bin/ariaTSsetup.py (hunk 1 above)
Comment (rzinke) [reply tools/bin/ariaTSsetup.py:487]: Fixed in https://github.com/aria-tools/ARIA-tools/pull/456

File: tools/ARIAtools/util/run_logging.py (hunk 2 above)
Comment (rzinke) [reply tools/ARIAtools/util/run_logging.py:14]: Addressed in https://github.com/aria-tools/ARIA-tools/pull/456/commits/c6f4ec46cc655ce0d20b0ae8e87b145dccdcf885

File: tools/ARIAtools/util/run_logging.py (hunk 3 above)
Comment (rzinke) [reply tools/ARIAtools/util/run_logging.py:17]: Addressed in https://github.com/aria-tools/ARIA-tools/pull/456/commits/c6f4ec46cc655ce0d20b0ae8e87b145dccdcf885

File: tools/ARIAtools/extractProduct.py (hunk 11 above)
Comment (rzinke) [reply tools/ARIAtools/extractProduct.py:533]: Agreed. Implemented in https://github.com/aria-tools/ARIA-tools/pull/456/commits/00ffde6bdd37aaa2feab53d8393e4d76831d2eb9

**Comment by rzinke** [comment]
This PR is closed due to conflicts from falling behind the dev branch, and because logging was separated into a separate PR (https://github.com/aria-tools/ARIA-tools/pull/456). It is replaced by https://github.com/aria-tools/ARIA-tools/pull/458.

**Pull request closed without merging**